### PR TITLE
Dispatch a change event only if the value changes

### LINKED ISSUE #4427: Don't trigger the event "change:resolution" on max/min zoom
#### Settings
I have listening the event `change:resolution` on an `ol.View` and also set the limits of zoom (`minZoom and maxZoom`).

#### Problem
When i zoom-out or zoom-in on the map until reach the limit, the event is triggered even if the zoom isn't changed

I made this example (https://jsfiddle.net/he1m2t00/) to show you the problem.

---

_If you want more descriptive issue or i'm wrong just tell me_

### PR DESCRIPTION
**Pull request by fredj**
Fixes #4427

### CODE REVIEW

**Comment by ahocevar** [comment]
Thanks @fredj. Please merge when Travis is happy.

**Comment by fredj** [comment]
A event will still be dispatched for different objects with the same value.
This is because the strict not equal operator is used:
```javascript
> [0, 1, 2] !== [0, 1, 2]
true
> {key: '42'} !== {key: '42'}
true
```

File: test/spec/ol/object.test.js [hunk 1]
```diff
@@ -168,6 +168,13 @@ describe('ol.Object', function() {
       expect(event.key).to.be('k');
     });
 
+    it('dispatches events only if the value is not different', function() {
```
Comment (jonataswalker) [inline test/spec/ol/object.test.js]: Shouldn't be *only if the value **is** different*?

Comment (fredj) [inline test/spec/ol/object.test.js]: fixed, thanks

**Comment by tschaub** [comment]
Looks good to merge.

**Pull request merged**